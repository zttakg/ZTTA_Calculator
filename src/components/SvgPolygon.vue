<template>
  <polygon
    :points="points"
    :x="strokeWidth"
    :y="strokeWidth"
    :fill="fillColor"
    :stroke-width="strokeWidth"
    :stroke="strokeColor"
  />
</template>

<script>
export default {
  name: "SvgPolygon",

  props: {
    points: String,
    strokeWidth: Number,
    strokeColor: String,
    fillColor: String
  }
};
</script>
