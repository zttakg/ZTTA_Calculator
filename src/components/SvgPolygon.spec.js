import { shallowMount } from "@vue/test-utils";
import SvgPolygon from "./SvgPolygon";

describe("<SvgPolygon /> component", () => {
  it("should render correct without props", () => {
    const wrapper = shallowMount(SvgPolygon);

    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should render all props", () => {
    const wrapper = shallowMount(SvgPolygon, {
      propsData: {
        points: "100 0, 0 200, 200 200, 100 0",
        strokeWidth: 5,
        strokeColor: "#000",
        fillColor: "#000"
      }
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
