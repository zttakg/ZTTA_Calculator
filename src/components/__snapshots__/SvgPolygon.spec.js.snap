// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SvgPolygon /> component should render all props 1`] = `<polygon points="100 0, 0 200, 200 200, 100 0" x="5" y="5" fill="#000" stroke-width="5" stroke="#000"></polygon>`;

exports[`<SvgPolygon /> component should render correct without props 1`] = `<polygon></polygon>`;
